pkg/peermgr/handle.go

```diff
@@ -320,3 +323,50 @@ func (swarm *Swarm) handleGetBlocksPack(s network.Stream, msg *pb.Message) error
 
 	return nil
 }
+
+func (swarm *Swarm) handleAskPierMaster(s network.Stream, data []byte) {
+	address := string(data)
+	if !swarm.piers.pierMap.hasPier(address) {
+		return
+	}
+	resp := &pb.CheckPierResponse{}
+	if swarm.piers.pierChan.checkAddress(address) {
+		resp.Status = pb.CheckPierResponse_HAS_MASTER
+	} else {
+		if swarm.piers.pierMap.checkMaster(address) {
+			resp.Status = pb.CheckPierResponse_HAS_MASTER
+		} else {
+			resp.Status = pb.CheckPierResponse_NO_MASTER
+		}
+	}
+	resp.Address = address
+	msgData, err := resp.Marshal()
+	if err != nil {
+		swarm.logger.Errorf("marshal ask pier master response: %s", err)
+		return
+	}
+	message := &pb.Message{
+		Data: msgData,
+		Type: pb.Message_CHECK_MASTER_PIER_ACK,
+	}
+	msg, err := message.Marshal()
+	if err != nil {
+		swarm.logger.Errorf("marshal response message: %s", err)
+		return
+	}
+
+	if err := swarm.p2p.AsyncSend(s.RemotePeerID(), msg); err != nil {
```

Comment:
Should be swarm.p2p.AsyncSend(s.RemotePeerID(), message)

Reply:
it is swarm.p2p.AsyncSend, not swarm.AsyncSend.